File: fortran/numbers/test/numbers_iterative_test.f90
program test_iterative
  use funit
  use numbers_iterative
  implicit none

  @test
  subroutine test_fibonacci()
    call assertEqual(fibonacci(5), 5)
  end subroutine

  @test
  subroutine test_factorial()
    call assertEqual(factorial(5), 120)
  end subroutine

  @test
  subroutine test_sum_numbers()
    call assertEqual(sum_numbers(5), 15)
  end subroutine

end program test_iterative

File: kotlin/numbers/src/test/kotlin/NumbersIterativeTest.kt
import kotlin.test.Test
import kotlin.test.assertEquals

class NumbersIterativeTest {

    private val nit = NumbersIterative()

    @Test
    fun testFibonacci() {
        assertEquals(5, nit.fibonacci(5))
    }

    @Test
    fun testFactorial() {
        assertEquals(120, nit.factorial(5))
    }

    @Test
    fun testSum_numbers() {
        assertEquals(15, nit.sumNumbers(5))
    }
}